pytest.ini

```diff
@@ -9,7 +9,7 @@ norecursedirs= build
 
 # Running all tests should take less than 12 minutes.
 # Otherwise, something went wrong.
-timeout = 720
+timeout = 1200
```

Comment:
any reason for keeping this as 20mins? What is the max time taken by the failing `test_warnings`?

Reply:
this is experimental, giving 3 times of the longest build run (~400 s), original 720s is around 2 times of the longest run.